### PR TITLE
Feature/error logging

### PR DESCRIPTION
**Pull request by jsundra**
`JsCallback` instances now log their own warnings/errors when `Apply` or `Invoke` is called. This provides much more informative errors for end users, including script names + line numbers. Callbacks internally handle messaging, so every single jsinterface doesn't need try/catch logic and matching string formatting.

Also, at a fundamental level `JsCallback`s never fail now. In the event of an error, `null` is returned. All of our usage of `JsCallback`s wrapped invokes with a try/catch that was only there to surface a warning/error and keep on executing. However if needed, `ExecutionError` is populated and can be checked after an invoke call to ensure it was successful.

Modules now have an optional name that can be set. Names do not have to be unique and are just used currently for informative log messages.

Maybe a slight performance increase or garbage reduction from an old `FIXME` getting fixed.

### CODE REVIEW

**Comment by jsundra** [comment]
This doesn't capture every use case `IJsCallback`s, specifically those created internally by jint. Going to shelve it for now until I can look into making the changes more widespread.

**Comment by jsundra** [comment]
Retooled this to not rely on module's as the source of script names. Instead script names are funneled through into the parser options, which propagates the name into Jint's internal `Location` object.

tl;dr - IJsCallbacks's JavaScriptExceptions now have names regardless of where they're generated from.

File: Orchid.Jint/JsCallback.cs [hunk 1]
```diff
@@ -69,8 +89,23 @@ public object Invoke(params object[] args)
                 jsArgs[i] = JsValue.FromObject(_context.Engine, args[i]);
             }
 
-            var result = _callback(jsThis, jsArgs);
-            return result.ToObject();
+            try
```
Comment (gdamron) [inline Orchid.Jint/JsCallback.cs:92]: seems like at least the callback invocation could be abstracted into a shared method.

**Review by gdamron — APPROVED** [review]
Looking forward to not wrapping every JS API callback in a try/catch

File: Orchid.Jint/JsCallback.cs (hunk 1 above)
Comment (jsundra) [reply Orchid.Jint/JsCallback.cs:92]: Ugh, yes. There is.... _a lot_ of this throughout jint. I want to clean that up as part of the ticket for allowing collections to be sent through callback invokes and the mess that forced spreading introduces :/

Comment (gdamron) [reply Orchid.Jint/JsCallback.cs:92]: 👍

**Review by ShannonKalei — APPROVED** [review]
